app/jenkins-for-jira-ui/src/components/JenkinsServerList/JenkinsServerList.tsx

```diff
@@ -23,19 +23,38 @@ const JenkinsServerList = (): JSX.Element => {
 	const analyticsClient = new AnalyticsClient();
 	const [jenkinsServers, setJenkinsServers] = useState<JenkinsServer[]>();
 	const [moduleKey, setModuleKey] = useState<string>();
+	const isMountedRef = React.useRef<boolean>(true);
+
 	const fetchAllJenkinsServers = async () => {
-		const servers = await getAllJenkinsServers() || [];
-		setJenkinsServers(servers);
+		try {
+			const servers = await getAllJenkinsServers() || [];
+			if (isMountedRef.current) {
+				setJenkinsServers(servers);
+			}
+		} catch (error) {
+			console.error('Error fetching Jenkins servers:', error);
+		}
 	};
 
 	const redirectToAdminPage = useCallback(async () => {
-		const currentModuleKey = await redirectFromGetStarted();
-		setModuleKey(currentModuleKey);
+		try {
+			const currentModuleKey = await redirectFromGetStarted();
+			if (isMountedRef.current) {
```

Comment:
/nit same as above,  looks like no point callin const currentModuleKey = await redirectFromGetStarted(); unless isMountedRef.current is true